include/dr/concepts/concepts.hpp

```diff
@@ -39,8 +39,6 @@ concept distributed_iterator =
     std::forward_iterator<I> &&
     requires(I &iter) {
       { lib::ranges::segments(iter) } -> rng::forward_range;
-      { lib::ranges::segment_index(iter) };
-      { lib::ranges::local_index(iter) };
```

Comment:
@BenBrock: I removed these so I could finish the PR. I did not remove all the code for `segment_index`.